From d9f2a079d0f4250ceabf50801b427229aa5edb15 Mon Sep 17 00:00:00 2001
From: Jason Wang <jasowang@redhat.com>
Date: Mon, 29 Sep 2014 00:59:13 -0400
Subject: [virt] hyperv: Add netpoll support

Message-id: <1411952358-34104-30-git-send-email-jasowang@redhat.com>
Patchwork-id: 96784
O-Subject: [RHEL7.1 PATCH 29/34] hyperv: Add netpoll support
Bugzilla: 1146357
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

From: Richard Weinberger <richard@nod.at>

Bugzilla: 1146357

In order to have at least a netconsole to debug kernel issues on
Windows Azure this patch implements netpoll support.
Sending packets is easy, netvsc_start_xmit() does already everything
needed.

Signed-off-by: Richard Weinberger <richard@nod.at>
Acked-by: Haiyang Zhang <haiyangz@microsoft.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 316158feff0078b266d6e423adb016d12eb96a5a)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/hyperv/netvsc_drv.c b/drivers/net/hyperv/netvsc_drv.c
index f6393ee..9e67ccd 100644
--- a/drivers/net/hyperv/netvsc_drv.c
+++ b/drivers/net/hyperv/netvsc_drv.c
@@ -749,6 +749,14 @@ static int netvsc_set_mac_addr(struct net_device *ndev, void *p)
  return err;
 }
 
+#ifdef CONFIG_NET_POLL_CONTROLLER
+static void netvsc_poll_controller(struct net_device *net)
+{
+ /* As netvsc_start_xmit() works synchronous we don't have to
+  * trigger anything here.
+  */
+}
+#endif
 
 static const struct ethtool_ops ethtool_ops = {
  .get_drvinfo = netvsc_get_drvinfo,
@@ -764,6 +772,9 @@ static const struct net_device_ops device_ops = {
  .ndo_validate_addr =  eth_validate_addr,
  .ndo_set_mac_address =  netvsc_set_mac_addr,
  .ndo_select_queue =  netvsc_select_queue,
+#ifdef CONFIG_NET_POLL_CONTROLLER
+ .ndo_poll_controller =  netvsc_poll_controller,
+#endif
 };
 
 /*
-- 
1.7.1